services/src/main/java/org/apache/unomi/services/services/SegmentServiceImpl.java

```diff
@@ -669,6 +676,9 @@ private void updateExistingProfilesForScoring(Scoring scoring) {
                 scriptParams.put("scoringValue", element.getValue());
                 for (Profile p : persistenceService.query(element.getCondition(), null, Profile.class)) {
                     persistenceService.updateWithScript(p.getItemId(), null, Profile.class, script, scriptParams);
+                    Event profileUpdated = new Event("profileUpdated", null, p, null, null, p, new Date());
+                    profileUpdated.setPersistent(false);
+                    eventService.send(profileUpdated);
                 }
```

Comment:
The event will be sent multiple times per profile (actualy, for each matching scoring element) - could you optimize and send it only once ?